### PR TITLE
JAMES-1778 VacationResponse should be marked as unactivated if outsid…

### PR DESCRIPTION
**Pull request by chibenwa**
…e of time range

### CODE REVIEW

**Comment by aduprat** [comment]
:+1:

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/GetVacationResponseMethodTest.java [hunk 1]
```diff
@@ -37,23 +38,29 @@
...
 import org.apache.james.jmap.model.VacationResponse;
 import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.util.date.ZonedDateTimeProvider;
 import org.junit.Before;
 import org.junit.Test;
 
 public class GetVacationResponseMethodTest {
 
     public static final String USERNAME = "username";
+    public static final ZonedDateTime ZONED_DATE_TIME_2014 = ZonedDateTime.parse("2014-09-30T14:10:00Z");
```
Comment (mbaechler) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/GetVacationResponseMethodTest.java]: _2014 is enough

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/GetVacationResponseMethodTest.java [hunk 2]
```diff
@@ -98,7 +106,84 @@ public void processShouldReturnTheAppropriateVacationResponse() {
...
+                    .enabled(false)
+                    .id(Vacation.ID)
+                    .textBody(Optional.of("I am in vacation"))
+                    .subject(Optional.of("subject"))
+                    .fromDate(Optional.of(ZONED_DATE_TIME_2014))
+                    .toDate(Optional.of(ZONED_DATE_TIME_2015))
+                    .build())
+                .build())
+            .build();
+        assertThat(result).containsExactly(expected);
```
Comment (mbaechler) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/GetVacationResponseMethodTest.java]: you just have to assert the enabled flag

**Comment by mbaechler** [comment]
Ok, I now understand why you did not agree about returning a modified enabled status. The server should not change this value because if you set a date in the future and you retrieve the server representation, it's not yet active thus it becomes disabled for the client ...
We need a read-only (response only) field for that.
@ddolcimascolo do you agree ?

**Comment by chibenwa** [comment]
@mbaechler I do not agree.
With what you propose, how you differenciate these two VacationResponse : 

```
                   {
                        "singleton" : {
                        "id": "singleton",
                        "isEnabled": "true",
                        "textBody": "Message explaining my wonderful vacations\"
                    }
```

Which is always disabled.... And : 

```
                   {
                        "singleton" : {
                        "id": "singleton",
                        "isEnabled": "false",
                        "textBody": "Message explaining my wonderful vacations\"
                    }
```

Which is always enabled as no dates means unlimited.

No, if you want to do it in some elegant way, you need to add one more attribute : isActive, that directly reflect the state of Vacation::isActive (both enabled and in date range). It is only set in the server response. This way we keep the immutable client set enabled property, and still have a way to know if the vacation is enabled.

But implies modifying JMAP specs again. What do you think ?

**Comment by chibenwa** [comment]
test this please

**Comment by mbaechler** [comment]
It's what I meant so I agree (:

**Comment by aduprat** [comment]
merged

**Pull request closed without merging**